drivers/input/input_esp32_touch_sensor.c

```diff
@@ -0,0 +1,335 @@
+			for (int i = 0; i < num_channels; i++) {
+				channel_cfg = &dev_cfg->channel_cfg[i];
+
+				/* Set interrupt threshold */
+				uint32_t benchmark_value;
+
+				touch_hal_read_benchmark(channel_cfg->channel_num,
+					&benchmark_value);
+				touch_hal_set_threshold(channel_cfg->channel_num,
+				(channel_cfg->channel_sens * benchmark_value / 100));
+			}
+		}
+		return;
+	}
+#endif /* defined(CONFIG_SOC_SERIES_ESP32) */
+
+	touch_hal_read_trigger_status_mask(&pad_status);
+#if defined(CONFIG_SOC_SERIES_ESP32)
+	touch_hal_clear_trigger_status_mask();
+#endif /* defined(CONFIG_SOC_SERIES_ESP32) */
+	for (int i = 0; i < num_channels; i++) {
+		uint32_t channel_status;
+
+		channel_cfg = &dev_cfg->channel_cfg[i];
+		channel_status = (pad_status >> channel_cfg->channel_num) & 0x01;
+
+#if defined(CONFIG_SOC_SERIES_ESP32)
+		if (channel_status != 0) {
+#elif defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+		uint32_t channel_num = (uint32_t)touch_hal_get_current_meas_channel();
+
+		if (channel_cfg->channel_num == channel_num) {
+#endif /* CONFIG_SOC_SERIES_ESP32 */
+			struct touch_sensor_channel_data
+				*channel_data = &dev_data->channel_data[i];
+
+			channel_data->status = channel_status;
+			(void)k_work_reschedule(&channel_data->work,
+						K_MSEC(dev_cfg->debounce_interval_ms));
+		}
+	}
+}
+
+static void rtc_isr(void *arg)
+{
+	uint32_t status = REG_READ(RTC_CNTL_INT_ST_REG);
+
+	if (arg != NULL) {
+		const struct device *dev = (struct device *)arg;
+		struct touch_sensor_data *dev_data = dev->data;
+
+		if (dev_data->rtc_intr_msk & status) {
+			touch_sensor_interrupt_cb(arg);
+		}
+	}
+
+	REG_WRITE(RTC_CNTL_INT_CLR_REG, status);
+}
+
+static esp_err_t rtc_isr_install(intr_handler_t intr_handler, void *handler_arg)
+{
+	esp_err_t err = ESP_OK;
+
+	REG_WRITE(RTC_CNTL_INT_ENA_REG, 0);
+	REG_WRITE(RTC_CNTL_INT_CLR_REG, UINT32_MAX);
+	err = esp_intr_alloc(ETS_RTC_CORE_INTR_SOURCE, 0, intr_handler, handler_arg, NULL);
+
+	return err;
+}
+
+/**
+ * Handle debounced touch sensor touch state.
+ */
+static void touch_sensor_change_deferred(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct touch_sensor_channel_data *channel_data =
+			CONTAINER_OF(dwork, struct touch_sensor_channel_data, work);
+	const struct device *dev = channel_data->dev;
+	const struct touch_sensor_data *dev_data = dev->data;
+	int key_index = channel_data - &dev_data->channel_data[0];
+	const struct touch_sensor_config *dev_cfg = dev->config;
+	const struct touch_sensor_channel_config
+		*channel_cfg = &dev_cfg->channel_cfg[key_index];
+
+#if defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+	if (channel_data->last_status != channel_data->status) {
+#endif /* defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3) */
+		input_report_key(dev, channel_cfg->zephyr_code,
+				channel_data->status, true, K_FOREVER);
```

Comment:
no release event?

Reply:
press and release events are informed by `channel_data->status`

Reply:
Sorry,  misread the sync bit for the value :-)